easybuild/tools/filetools.py

```diff
@@ -266,10 +294,43 @@ def download_file(filename, url, path):
     basedir = os.path.dirname(path)
     mkdir(basedir, parents=True)
 
+    protocol = url.split(':')[0]
+    if protocol not in PROTOCOL_MAP:
+        _log.exception("can't handle url: %s, unsupported protocol %s currently only %s are supported" %
+                       (url, protocol, PROTOCOL_MAP.keys()))
+    _log.debug("Downloading %s from %s to %s using protocol %s", filename, url, path, protocol)
+    # call the download_XXX function
+    return getattr(sys.modules[__name__], "download_%s" % PROTOCOL_MAP[protocol])(filename, url, path, timeout)
```

Comment:
eek!

why not use the functions themselves as values in `PROTOCOL_MAP` instead?